### PR TITLE
Implement replies and new topics in Talk pages.

### PR DESCRIPTION
**Pull request by dbrant**

### CODE REVIEW

File: app/src/main/java/org/wikipedia/talk/TalkTopicActivity.kt [hunk 1]
```diff
@@ -162,10 +223,119 @@ class TalkTopicActivity : BaseActivity() {
...
+        }
+
+        // if the message is not signed, then sign it explicitly
+        if (!body.endsWith("~~~~")) {
+            body += " ~~~~"
+        }
+        if (!isNewTopic()) {
+            // add two explicit newlines at the beginning, to delineate this message as a new paragraph.
+            body = "\n\n" + body
+        }
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/talk/TalkTopicActivity.kt:260]: If there's no replies in a new topic, the first reply will stick with the topic.

File: app/src/main/java/org/wikipedia/talk/TalkTopicActivity.kt [hunk 2]
```diff
@@ -162,10 +223,119 @@ class TalkTopicActivity : BaseActivity() {
         }
     }
 
+    inner internal class ReplyTextWatcher: TextWatcher {
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/talk/TalkTopicActivity.kt:226]: `inner` can be removed.

**Review by cooltey — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/talk/TalkTopicActivity.kt (hunk 2 above)
Comment (dbrant) [reply app/src/main/java/org/wikipedia/talk/TalkTopicActivity.kt:229]: it cannot.

File: app/src/main/java/org/wikipedia/talk/TalkTopicActivity.kt (hunk 1 above)
Comment (dbrant) [reply app/src/main/java/org/wikipedia/talk/TalkTopicActivity.kt:263]: should be fixed.

File: app/src/main/java/org/wikipedia/talk/TalkTopicActivity.kt (hunk 2 above)
Comment (cooltey) [reply app/src/main/java/org/wikipedia/talk/TalkTopicActivity.kt:229]: You're correct. Would it be possible to add `@Suppress("RedundantInnerClassModifier")` for it?
```
    @Suppress("RedundantInnerClassModifier")
    internal inner class ReplyTextWatcher: TextWatcher {
```

Comment (dbrant) [reply app/src/main/java/org/wikipedia/talk/TalkTopicActivity.kt]: yep

**Review by cooltey — APPROVED** [review]
LGTM. @sharvaniharan Please review and merge if it looks good to you too.

**Comment by cooltey** [comment]
@sharvaniharan I just merged it. Please create a follow-up if you see issues, thanks!

**Pull request merged**